docs/api/usb/USBCDC_ECM.md

```diff
@@ -0,0 +1,52 @@
+# USBCDC_ECM
+
+<span class="images">![](https://os.mbed.com/docs/mbed-os/development/mbed-os-api-doxy/class_u_s_b_c_d_c___e_c_m.png)<span>USBCDC_ECM class hierarchy</span></span>
+
+The USBCDC_ECM class emulates an Ethernet interface over USB. You can use the interface to send and receive Ethernet frames over USB.
+
+## USBCDC_ECM class reference
+
+[![View code](https://www.mbed.com/embed/?type=library)](https://os.mbed.com/docs/mbed-os/development/mbed-os-api-doxy/class_u_s_b_c_d_c___e_c_m.html)
+
+<span class="notes">**Note:** Because Windows doesn’t provide native support for the USB CDC-ECM model, you must use third party drivers to use this class on Windows.</span>
+
+## USBCDC_ECM example
+
+The example below sends an Ethernet frame that carries "Hello world" payload with a custom EtherType to the host PC. You can capture the frame by using a program called "Wireshark":
+
+1. Flash the board, and ensure the target's USB is plugged into the PC.
+2. Open Wireshark.
+3. Click **Capture > Options** to select the correct capture interface.
+4. Click **Capture > Start**.
+5. Click captured packet from source address 12:34:56:78:9a:bc to see the "Hello world" payload.
+
+```C++ TODO
```

Comment:
Are there plans to make this an importable example? If there's no set plan for this, please add this example to this repository: https://github.com/ARMmbed/mbed-os-examples-docs_only

Reply:
Do you know what are the plans for other USB device examples? I can add the example if needed.

Reply:
We would like to make the other USB device examples importable examples instead of having them hardcoded in the documentation.

Reply:
PR created https://github.com/ARMmbed/mbed-os-examples-docs_only/pull/50